### PR TITLE
11.0 website multi theme port from 10

### PR DESCRIPTION
**Pull request by yelizariev**
<s>WIP
We will check it internally first. </s>

### CODE REVIEW

**Comment by emagdalenaC2i** [comment]
Is not migrated already? Check https://github.com/OCA/website/pull/398

**Comment by pedrobaeza** [comment]
This is a recent improvement.

**Review by Railia — APPROVED** [review]
👍

**Comment by yelizariev** [comment]
@emagdalenaC2i #398 was made by me :wink:

**Comment by yelizariev** [comment]
I'll add more updates from 10.0 after https://github.com/OCA/website/pull/453 and  https://github.com/OCA/website/pull/452

**Comment by yelizariev** [comment]
New updates to be merged here: <s>https://github.com/OCA/website/pull/457</s> https://github.com/OCA/website/pull/456

**Comment by yelizariev** [comment]
This error is randomly shown on travis checks
```
ssertionError: ("TypeError: undefined is not an object (evaluating 'manipulatorOffset.top')\n"
' at '
'http://127.0.0.1:8069/web/content/512-6922043/web_editor.assets_editor.js:132 '
'(in cover)\n'
' at '
'http://127.0.0.1:8069/web/content/512-6922043/web_editor.assets_editor.js:133 '
'(in toggleFocus)\n'
' at '
'http://127.0.0.1:8069/web/content/512-6922043/web_editor.assets_editor.js:162\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:802\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:796 '
'(in fire)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:797 '
'(in add)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:802\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:625 '
'(in each)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:802\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:803 '
'(in Deferred)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:802 '
'(in then)\n'
' at '
'http://127.0.0.1:8069/web/content/512-6922043/web_editor.assets_editor.js:162 '
'(in _activateSnippet)\n'
' at '
'http://127.0.0.1:8069/web/content/512-6922043/web_editor.assets_editor.js:150\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:892 '
'(in dispatch)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:865 '
'(in handle)\n'
' at :0 (in dispatchEvent)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:4187 '
'(in trigger_mouse_event)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:4187 '
'(in _click)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:4187 '
'(in _text)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:4186 '
'(in text)\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:4204\n'
' at :0\n'
' at http://127.0.0.1:8069/web/content/509-3597dce/web.assets_common.js:44\n'
'(leaf frame on top)')

```

**Comment by yelizariev** [comment]
I'd like to set status "Needs review" to this issue and postpone with #456 which is not merged yet

**Review by Railia — APPROVED** [review]

**Comment by yelizariev** [comment]
now #456 is in this PR too, i.e. all updates from 10.0 are here

**Comment by yelizariev** [comment]
Checked internally and found misunderstanding about Multi-footer. I've added a note to documentation in latest commit

**Review by ilmir-k — APPROVED** [review]
tested :+1: 
looks good to me

**Review by yajo — APPROVED** [review]
@yelizariev Is this ready to merge then?

**Comment by yelizariev** [comment]
@Yajo yes, it's ready. We did all we can. We will continue to test it with ``website_multi_company`` module once this PR is merged

**Comment by pedrobaeza** [comment]
Please squash a bit some unneeded commits and I'll merge. Thanks for the work checking these rough edges.

**Comment by yelizariev** [comment]
I found and done another thing that was in discussion https://github.com/OCA/website/pull/409#issuecomment-361230854

**Comment by yelizariev** [comment]
@pedrobaeza most commits were cherry-picked. I think it's better to keep them to easy comparing with 10.0 branch.

File: website_multi_theme/data/ir_module_category.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 Ivan Yelizariev <https://it-projects.info/team/yelizariev>
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html). -->
+<odoo>
+
+    <record model="ir.module.category" id="base.module_category_theme">
+        <field name="exclusive" eval="False"/>
```
Comment (yajo) [inline website_multi_theme/data/ir_module_category.xml:7]: I can't believe it was so simple! Could you backport this to v10? :thinking: Or was it hardcoded in v10?

**Review by yajo — APPROVED** [review]
Code & functional review

File: website_multi_theme/data/ir_module_category.xml (hunk 1 above)
Comment (yelizariev) [reply website_multi_theme/data/ir_module_category.xml:7]: Yes, it's hardcoded and the only way is rewrite (not inherit) one method

**Pull request merged**